pkg/minishift/addon/addon_meta_test.go

```diff
@@ -110,8 +110,73 @@ func Test_required_vars_empty_if_not_specified(t *testing.T) {
 	testMap["Description"] = []string{"Acme Add-on"}
 
 	addOnMeta := getAddOnMeta(testMap, t)
+	requiredVars, _ := addOnMeta.RequiredVars()
+	minishiftTesting.AssertEqualSlice(requiredVars, []string{}, t)
+}
+
+func Test_var_defaults_empty_if_not_specified(t *testing.T) {
+	testMap := make(map[string]interface{})
+	testMap["Name"] = "acme"
+	testMap["Description"] = []string{"Acme Add-on"}
+
+	addOnMeta := getAddOnMeta(testMap, t)
+	varDefaults, _ := addOnMeta.VarDefaults()
+	if len(varDefaults) != 0 {
+		t.Fatal(fmt.Sprintf("Expected empty var default, but got: '%s'", len(varDefaults)))
+	}
+}
+
+func Test_var_defaults_meta_is_extracted(t *testing.T) {
+	testMap := make(map[string]interface{})
+	testMap["Name"] = "acme"
+	testMap["Description"] = []string{"Acme Add-on"}
+	expectedDefaultEnv := "USER=foo"
+	testMap["Var-Defaults"] = expectedDefaultEnv
+
+	addOnMeta := getAddOnMeta(testMap, t)
+	varDefaults, _ := addOnMeta.VarDefaults()
+	if len(varDefaults) != 1 {
+		t.Fatal(fmt.Sprintf("Expected one var default, but got: '%s'", len(varDefaults)))
+	}
+}
+
+type varDefaultTestCase struct {
+	expectedVarDefault string
+	expectedError      string
+}
+
+func Test_invalid_var_defaults_meta(t *testing.T) {
+	testCases := []varDefaultTestCase{
+		{
+			expectedVarDefault: "USER=foo,,",
```

Comment:
What's about something like 'User=foo=bar'? I'd like to see a test for this as well.

Reply:
What's about the empty string?

    Var-Defaults:

Reply:
Done